### PR TITLE
Fix X11 WaitEventsIterator busy loop

### PR DESCRIPTION
**Pull request by jwilm**
Certain events were not being checked for in the PollEventsIterator
and would be left in the X11 queue forever. XPeekEvent in the
WaitEventsIterator would never block because an item was still in the
queue causing a busy loop. The problem is solved by using XNextEvent
in PollEventsIterator with a check to make sure it will never block.

I added the log crate for debugging, but realize now that you might not want it in glutin. Lmk if you want it removed or feel free to patch on top of this.

### CODE REVIEW

File: src/api/x11/window.rs [hunk 1]
```diff
@@ -181,18 +181,29 @@ impl<'a> Iterator for PollEventsIterator<'a> {
...
+            // types of out the queue to prevent busy looping when there's an
+            // event of another type glutin doesn't care about.
+            unsafe {
+                let count = (xlib.XPending)(self.window.x.display.display);
+                if count == 0 {
+                    return None;
                 }
-            }
+
+                let res = (xlib.XNextEvent)(
```
Comment (tomaka) [inline src/api/x11/window.rs]: Nit: everything should be on the same line.

**Comment by tomaka** [comment]
Thanks for the PR.

Please remove the `log` and `bitflags` crate however.

**Comment by jwilm** [comment]
Oh, didn't realize bitflags wasn't for this patch :laughing:. I originally built this on top of another branch with some other features..

Anyway, fixed both the nit and the extra deps. Should be good to go.

**Comment by jwilm** [comment]
Should I redirect this PR over to winit?

**Comment by tomaka** [comment]
@jwilm Yes please

**Pull request closed without merging**